docs/docs/enclave-configuration.md

```diff
@@ -316,184 +316,72 @@ The path should be absolute or relative to the root of the enclave module.
     the usual backslashes.
 
-## Assisted Configuration of Native Image Builds
+## Assisted configuration of native image builds
 
-One way of generating the reflection and serialization configuration files is by using
-[native-image-agent](https://www.graalvm.org/reference-manual/native-image/BuildConfiguration/#assisted-configuration-of-native-image-builds).
-The agent will track usage of dynamic features and generate the configuration files when run
-against a regular Java VM.
+You can generate the reflection and serialization configuration files by using the
+[native-image-agent](https://www.graalvm.org/22.0/reference-manual/native-image/Agent/#assisted-configuration-of-native-image-builds).
+This command will track the use of dynamic features and generate the configuration files when run against a regular 
+Java VM.
 
-To ensure all the necessary classes/resources are included in the configuration files you should ensure
-all enclave code paths are executed, for example by writing extensive tests and running them in `mock` mode.
+To include all the necessary classes and resources in the generation of the configuration files and for Gradle to 
+pick up all the configuration files:
 
-When placed under `enclave/src/main/resources/META-INF/native-image` the configuration files will be picked up
-by Gradle when building the `enclave` in `simulation`, `debug` or `release` modes.
+* Execute all the enclave code paths by running extensive tests in `mock mode` with flags set up to place the 
+  configuration files in the path: `enclave/src/main/resources/META-INF/native-image`.
+* Build the enclave in `simulation` mode.
 
-Running the host through Gradle and/or JUnit tests while the agent is enabled will likely cause Gradle, JUnit
-or host classes to be present in the configuration files. To avoid this you can configure
-[filters](https://www.graalvm.org/reference-manual/native-image/BuildConfiguration/#agent-advanced-usage)
-to ensure host code is excluded or you can run the host as an executable JAR and trigger enclave logic
-by running it normally, like you would if you were to deploy it, for example, by sending requests from the
-host and/or client triggering as much of the enclave logic as possible.
-You can use the [Shadow Gradle plugin](https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow)
-to create an executable JAR which contains all the host's and enclave's dependencies.
+Later on, Gradle will automatically pick up all the configuration files when you run the enclave in any mode.
 
-It is possible the generated configuration files aren't always correct, meaning you may need to adjust filters
-and/or edit and maintain them throughout the development process. We recommend keeping these files
-version controlled to make maintenance easier.
+You might need to adjust filters and edit the generated configuration files throughout development.
```

Comment:
I think this is a bit too vague compared to before. It still makes sense to tell the user how to add filters to avoid some noise, link the Graal page and write a description on how to do that. I would keep the sentence 
`Running the host through Gradle and/or JUnit tests while the agent is enabled will likely cause Gradle, JUnit
or host's classes to be present in the configuration files.
To avoid this and ensure that only enclave related classes and resources are included, you can configure
[filters](https://www.graalvm.org/reference-manual/native-image/BuildConfiguration/#agent-advanced-usage).
If you are not running the agent with tests, 
you can run the host as an executable JAR and  trigger as much enclave logic as possible by sending requests from the
host and/or client.

To create an executable Jar you can use the  [Shadow Gradle plugin](https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow).
You might need to adjust the generated configuration files and the filters few times and edit them throughout the development process`

`

Reply:
Rewrote.